python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -271,28 +313,222 @@ def latest_materialization_records_by_key(
 
     @public
     def materialization_records_for_key(
-        self, asset_key: AssetKey, limit: int
+        self, asset_key: AssetKey, limit: int, after_cursor_partition: Optional[bool] = True
     ) -> Iterable["EventLogRecord"]:
         """Fetches asset materialization event records for asset_key, with the earliest event first.
 
         Args:
             asset_key (AssetKey): The asset to fetch materialization events for
             limit (int): The number of events to fetch
+            after_cursor_partition (Optional[bool]): If True, only materializations with partitions
+                after the cursor's current partition will be returned. By default, set to True. For
+                unpartitioned assets, this parameter is ignored.
         """
         from dagster._core.events import DagsterEventType
         from dagster._core.storage.event_log.base import EventRecordsFilter
 
-        cursor_dict = json.loads(self.cursor) if self.cursor else {}
+        asset_key = check.inst_param(asset_key, "asset_key", AssetKey)
+        if asset_key not in self._assets_by_key:
+            raise DagsterInvalidInvocationError(f"Asset key {asset_key} not monitored by sensor.")
+
+        _, cursor = self._get_cursor(asset_key)
+
+        partitions_to_fetch = (
+            self._get_partitions_after_cursor(asset_key) if after_cursor_partition else None
+        )
+
         return self.instance.get_event_records(
             EventRecordsFilter(
                 event_type=DagsterEventType.ASSET_MATERIALIZATION,
                 asset_key=asset_key,
-                after_cursor=cursor_dict.get(str(asset_key)),
+                after_cursor=cursor,
+                asset_partitions=partitions_to_fetch,
             ),
             ascending=True,
             limit=limit,
         )
 
+    def _get_cursor(self, asset_key: AssetKey):
+        unpacked_cursor = json.loads(self.cursor) if self.cursor else {}
+        partition_key, cursor = unpacked_cursor.get(str(asset_key), (None, None))
+        return partition_key, cursor
+
+    @public
+    def latest_materialization_by_partition(
+        self,
+        asset_key: AssetKey,
+        after_cursor_partition: Optional[bool] = True,
+    ) -> Mapping[str, Optional["EventLogRecord"]]:
+        """
+        Fetches the most recent materialization event record for each partition of each asset.
```

Comment:
it took me a minute to figure out why this fn was different than `latest_materiazations_by_key`, so we could probably make the doc string comments more explanatory. Maybe an example of what get returned, but not sure yet what that would look like

Reply:
Materializations should be plural to match our convention for naming dictionaries.  Also, if this is returning EventLogRecords, it should be `latest_materialization_records_by_partition`.